pkg/tekton/pipelines.go

```diff
@@ -493,6 +496,33 @@ func ApplyPipeline(jxClient versioned.Interface, tektonClient tektonclient.Inter
 	return nil
 }
 
+// StructureForPipelineRun finds the PipelineStructure for the given PipelineRun, trying its name first and then its
+// Pipeline name, returning an error if no PipelineStructure can be found.
+func StructureForPipelineRun(jxClient versioned.Interface, ns string, run *pipelineapi.PipelineRun) (*v1.PipelineStructure, error) {
+	structure, err := jxClient.JenkinsV1().PipelineStructures(ns).Get(run.Name, metav1.GetOptions{})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Try again but with the Pipeline name for this run.
+			pipelineName := run.Labels[pipeline.GroupName+pipeline.PipelineLabelKey]
+			// Fall back on the PipelineRef.Name if there isn't a label.
+			if pipelineName == "" {
+				pipelineName = run.Spec.PipelineRef.Name
+			}
+			// If we still have no name, error out.
+			if pipelineName == "" {
+				return nil, fmt.Errorf("couldn't find a Pipeline name for PipelineRun %s", run.Name)
+			}
+			structure, err = jxClient.JenkinsV1().PipelineStructures(ns).Get(pipelineName, metav1.GetOptions{})
```

Comment:
If we know that lookup by pipeline run name can fail and the "better" option is by pipeline name, why don't we do the lookup always by the latter. Less API calls in average and less nested logic.

Reply:
Done. =)